### PR TITLE
query for panels based on panel set id

### PR DESCRIPTION
**Pull request by wct9667**
### Description
Added a query for `getPanelsFromPanelSetID` that returns an array of panels `{id, image, index}` based on a given `panel_set_id`.

Example Output:
image

### CODE REVIEW

File: src/database/services/PanelService.ts [hunk 1]
```diff
@@ -55,6 +61,23 @@ class PanelService {
...
     }
+
+    /**
+     * Get all panels that are associated with a specific panelSet
+     * @param {number} panel_set_id ID of panelSet
+     * @returns {object[]} An array of objects with id, image, index properties
+     */
+    static async getPanelsFromPanelSetID(panel_set_id: number){
+        // Find all panels on requested panelSet 
+        const panels = await Panel?.findAll({where: {panel_set_id: panel_set_id}});
```
Comment (tstodter) [inline src/database/services/PanelService.ts]:
```suggestion
        const panels = await Panel.findAll({where: {panel_set_id}});
```
Can also be this. I don't think the `?.` is necessary, safe to assume Panel exists. Also there's some easy sugar there.

File: src/database/services/PanelService.ts [hunk 2]
```diff
@@ -55,6 +61,23 @@ class PanelService {
...
+        if(!(panels?.length>0)) return undefined
```
Comment (tstodter) [inline src/database/services/PanelService.ts]: Suggest just returning `[]` instead of undefined.

File: src/database/services/PanelService.ts [hunk 3]
```diff
@@ -55,6 +61,23 @@ class PanelService {
...
+        
+        //Map panels to keep only needed data
+        const parsedPanels = panels.map((p)=>{return{id: p.id, image: p.image, index: p.index}});
```
Comment (tstodter) [inline src/database/services/PanelService.ts]:
```suggestion
        const parsedPanels = panels.map((p) => ({
          id: p.id, 
          image: p.image, 
          index: p.index
        }))
```

**Review by nca6109 — APPROVED** [review]
Looks good

**Pull request merged**